autopush/senderids.py

```diff
@@ -40,63 +41,73 @@
 
 class SenderIDs(object):
     """Handle Read, Write and cache of SenderID values from S3"""
-    _updated = 0
     _expry = SENDERID_EXPRY
     _senderIDs = {}
     _use_s3 = True
     KEYNAME = "senderids"
+    service = None
 
     def __init__(self, args):
         """Optionally load or fetch the set of SenderIDs from S3"""
         self.conn = S3Connection()
-        self.ID = args.get("s3_bucket", DEFAULT_BUCKET)
+        self.ID = args.get("s3_bucket", DEFAULT_BUCKET).lower()
         self._expry = args.get("senderid_expry", SENDERID_EXPRY)
         self._use_s3 = args.get("use_s3", True)
         senderIDs = args.get("senderid_list", {})
+        self.service = None
         if senderIDs:
             if type(senderIDs) is not dict:
                 log.err("senderid_list is not a dict. Ignoring")
             else:
+                # We're initializing, so it's ok to block.
                 self.update(senderIDs)
```

Comment:
Normally, I'm not either, however in this case, it's kinda important that we have data that needs to be written and successfully before processing starts. Ben noted early on that if there's a failure, it needs to happen very early on so that it's caught by ops. Otherwise, there's the risk that it doesn't get detected.